pilot/pkg/model/jwks_resolver.go

```diff
@@ -259,20 +264,22 @@ func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, erro
 		lastRefreshedTime: now,
 		lastUsedTime:      now,
 	})
-
+	if err != nil {
+		// fetching the public key in the background
+		jwksuriChannel <- key
```

Comment:
It can be done in concurrent execution, but in that case we need to change `r.refreshInterval` (which is 20 mins rn) to some smaller value to make sure that our ticker executes instantly. If we do that there might be other cases which could go wrong.

Reply:
what type of edge cases you're talking about ?

Reply:
oops, misunderstood ticker. Replace what I said with `r.refreshTicker.C <- time.Now()`.

The issue would be if the refreshTicker select block is running and slow (it could be making 100s of http calls). We do the send on jwksuriChannel >5 times, it will block until the refreshTicker part is done.

Alternatively we can just have the on-demand async calls decoupled entirely from the refreshTicker (and probably just refresh that 1 url, not loop over all of them

Reply:
Actually the single call may be better, otherwise for every new jwks we need to refreshall other jwks - may not scale well

Reply:
> otherwise for every new jwks we need to refreshall other jwks - may not scale well

yes, we should address this.

The other way could be to update the refresh() logic so that it only fetches the jwks URI that are newly added from here.

Reply:
I have discussed with him, since we don't have many `jwksuri` in the cache in one single time. I will add a new logic in my follow-up work where we will just try to fetch the pubkey only for the jwksuri which is failing.

Reply:
Could you just do it in this PR? I'm a bit worried about the performance even if the customer is only using a small number of `jwksUri` because each of the `jwksUri` will trigger a new config push right?

Reply:
even after we refresh one single `jwksuri`, it is gonna create a new config push and will push the whole cache again. The only drawback which I see in our current solution is that it will try to refresh all the `jwksuri` in the cache. 
@howardjohn do we have any option to push one single `jwksuri` - `pubkey` to envoy ?